contracts/burners/XSushiBurner.vy

```diff
@@ -0,0 +1,174 @@
+# @version 0.2.12
+"""
+@title xSushi Burner
+@notice Leave SushiBar and send sushi to receiver
+"""
+
+from vyper.interfaces import ERC20
+
+
+interface xSushi:
+    def leave(_share:uint256): nonpayable
+
+receiver: public(address)
+recovery: public(address)
+owner: public(address)
+is_killed: public(bool)
+emergency_owner: public(address)
+future_owner: public(address)
+future_emergency_owner: public(address)
+sushi: public(address)
+
+
+event Burn:
+    lp_token: address
+    amount: uint256
+    token0_amount:uint256
+
+@external
+def __init__(_receiver:address, _recovery: address, _owner: address, _emergency_owner: address):
+    """
+    @notice Contract constructor
+    @param _receiver the receiver address to which the resultant tokens will be sent
+    @param _recovery Address that tokens are transferred to during an
+                     emergency token recovery.
+    @param _owner Owner address. Can kill the contract, recover tokens
+                  and modify the recovery address.
+    @param _emergency_owner Emergency owner address. Can kill the contract
+                            and recover tokens.
+    """
+    self.receiver = _receiver
+    self.recovery = _recovery
+    self.owner = _owner
+    self.emergency_owner = _emergency_owner
+    self.sushi = 0x6B3595068778DD592e39A122f4f5a5cF09C90fE2
+
+
+@external
+def burn(_coin: address) -> bool:
+    """
+    @notice Convert `_coin` by leaving Sushibar and get Sushi, then send it to receiver
+    @param _coin Address of the coin being converted
+    @return bool success
+    """
+    assert not self.is_killed  # dev: is killed
+    
```

Comment:
should assert _coin is xSUSHI: `0x8798249c2E607446EfB7Ad49eC89dD1865Ff4272`
in case some suspicious contract which implement `.leave()` and do some attack

Reply:
noted. tks